service/history/workflowTaskHandler.go

```diff
@@ -114,6 +115,7 @@ func newWorkflowTaskHandler(
 		namespaceCache: namespaceCache,
 		metricsClient:  metricsClient,
 		config:         config,
+		historyEngine:  historyEngine,
```

Comment:
if you move the mutable state function creation out of history engine, then ref to history engine is not necessary
https://github.com/temporalio/temporal/pull/1745/files#r693266028